compiler/rustc_lexer/src/lib.rs

```diff
@@ -613,7 +611,9 @@ impl Cursor<'_> {
         }
     }
 
-    fn number(&mut self, first_digit: char) -> LiteralKind {
+    /// Parses a number and in `.1` returns the offset of the literal suffix
+    /// (this will be at the end of the token if there is no suffix)
+    fn number(&mut self, first_digit: char) -> (LiteralKind, u32) {
```

Comment:
```suggestion
    fn number(&mut self, first_digit: char) -> (LiteralKind, Option<u32>) {
```
It would be simpler to return the offset only if it's overridden to something other than the default value of `self.pos_within_token()`.